storage/cloud.py

```diff
@@ -696,6 +696,28 @@ def cancel_chunked_upload(self, uuid, storage_metadata):
         for chunk in self._chunk_list_from_metadata(storage_metadata):
             self.remove(chunk.path)
 
+    def clean_partial_uploads(self, deletion_date_threshold):
+        self._initialize_cloud_conn()
+        path = self._init_path("uploads")
+        paginator = self.get_cloud_conn().get_paginator(self._list_object_version)
+        for page in paginator.paginate(Bucket=self._bucket_name, Prefix=path):
+            for obj_info in page.get("Contents", []):
+                if obj_info["LastModified"] <= datetime.now(timezone.utc) - deletion_date_threshold:
+                    obj = self.get_cloud_bucket().Object(obj_info["Key"])
+                    try:
+                        obj.load()
+                        obj.delete()
+                        logger.debug(
+                            "Expired blob removed from uploads folder: %s", obj_info["Key"]
+                        )
+                    except botocore.exceptions.ClientError as s3r:
+                        if not s3r.response["Error"]["Code"] in _MISSING_KEY_ERROR_CODES:
+                            logger.debug(
```

Comment:
`logger.exception` might be more appropriate here.

Reply:
Agreed